exporter.go

```diff
@@ -898,13 +898,12 @@ func (e *Exporter) extractSlowLogMetrics(ch chan<- prometheus.Metric, c redis.Co
 }
 
 func (e *Exporter) extractLatencyMetrics(ch chan<- prometheus.Metric, c redis.Conn) {
-	if reply, err := doRedisCmd(c, "LATENCY", "LATEST"); err == nil {
-		var eventName string
-		if tempVal, _ := reply.([]interface{}); len(tempVal) > 0 {
-			latencyResult := tempVal[0].([]interface{})
-			var spikeLast, spikeDuration, max int64
-			if _, err := redis.Scan(latencyResult, &eventName, &spikeLast, &spikeDuration, &max); err == nil {
-				spikeDurationSeconds := float64(spikeDuration) / 1e3
+	if reply, err := redis.Values(doRedisCmd(c, "LATENCY", "LATEST")); err == nil {
+		for _, l := range reply {
+			if latencyFields, err := redis.Values(l.([]interface{}), nil); err == nil {
+				eventName := (string)(latencyFields[0].([]byte))
```

Comment:
I prefer the use of `redis.Scan()` over directly casting the variables as it error checks and is safer.

Reply:
I think I misunderstood what you meant by "use another call to redis.Values() (with err check) in line 905 instead of the current cast".  
Is this more of what you had in mind?

Reply:
Yup, this looks good, thanks!